### ISSUE TITLE
[BUG] UI Mode, status line summary incorectly computed when running tests what depend form other tests (projects)

### ISSUE THREAD

**Issue by jaktestowac** [microsoft/playwright#27365]
tl;dr
When running project dependent tests summary (status line) is incorrectly computed:
![obraz](https://github.com/microsoft/playwright/assets/72373858/4f14cd99-ca50-4df0-8dd8-c5c4f1d9640c)

### System info
- Playwright Version: [v1.38.1]
- Operating System: [Windows 11]
- Browser: [Chromium]
- Other info: All needed code attached

### Source code

- [x ] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
// playwright.config.ts
import { defineConfig } from "@playwright/test";

export default defineConfig({
  projects: [
    {
      name: "setup",
      grep: /@setup/,
    },
    {
      name: "chromium",
      grep: /@chromium/,
      dependencies: ["setup"],
    },
  ],
});
```

**Test file (self-contained)**

```js
// tests/example.spec.ts
import { expect, test } from "@playwright/test";

test("run @setup", async ({ page }) => {
  console.log("Test setup executed");
});

test("run @chromium", async ({ page }) => {
  console.log("Test chromium executed");
});
```

**Steps**
- Start UI mode: npx playwright test --ui
- Execute test `run @chromium` (this test trigger @setup test)

**Expected**

Count of tests will be computed correctly in UI Mode:
2/2 passed (100%)

**Actual**
2/1 passed (200%)
